### PR TITLE
provider-local: Update outdated comment for appending hosts.toml files

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
After https://github.com/gardener/gardener/pull/8834, provider-local appends hosts.toml files to the `provision` OSC. The registry-cache:
- for the registry-mirror case: continues to append them to the `reconcile` OSC
- for the registry-cache cache: appends unit and files are created by the unit when the cache endpoint is ready

Hence, provider-local no longer needs to use the custom `appendFileIfNotPresent` func/behaviour. 

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
N/A

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/provider-local/webhook/controlplane/ensurer.go [hunk 1]
```diff
@@ -89,11 +89,12 @@ func (e *ensurer) EnsureAdditionalProvisionFiles(_ context.Context, _ extensions
...
-		// appendFileIfNotPresent in used instead of appendUniqueFile intentionally to allow enabling and testing the registry-cache extension in local setup.
-		// A file appended by the registry-cache extension is always picked up because:
-		// - if a file is already appended by the registry-cache extension, provider-local won't overwrite it (appendFileIfNotPresent)
-		// - if a file is already appended by provider-local, the registry-cache extension will overwrite it (appendUniqueFile)
-		appendFileIfNotPresent(new, extensionsv1alpha1.File{
+		// provider-local hosts.toml files are added to the OperatingSystemConfig with purpose "provision".
+		// registry-cache extension hosts.toml files are added to the OperatingSystemConfig with purpose "reconcile".
+		// Hence, there is no conflict between provider-local and registry-cache.
+		// hosts.toml files added by the registry-cache extension should always overwrite the provider-local ones
+		// to allow testing the extension in local setup.
```
Comment (Kostov6) [inline pkg/provider-local/webhook/controlplane/ensurer.go:96]:
```suggestion
		// provider-local assumes that it's the only webhook acting on OSC with provision purpose. If a webhook wants to work with provider-local and wants to overwrite hosts.toml it should act on OSC with puprose reconcile
```
This wording creates an implicit dependency on registry-cache in g/g's code just by referencing it. I suggest we use a more general comment that enforces the needed assumptions in a broader context. By doing so we enforce external actors to adapt to g/g and not vice-versa while still not restricting registry-cache's functionality. Your proposed comment is more suited for [registry-cache's ensurer](https://github.com/gardener/gardener-extension-registry-cache/blob/2552b21038b017ff3ca877fd2d071d17f3aae980/pkg/webhook/mirror/ensurer.go#L106) imo. Wdyt?

**Review by Kostov6 — COMMENTED** [review]
Thanks for the PR! Just a minor suggestion from my side

**Review by timuthy — CHANGES_REQUESTED** [review]
This for the PR.

However, having https://github.com/gardener/gardener/pull/10050 in mind, this comment is removed anyways and the registry cache extension is supposed to contribute to `.spec.criConfig.containerd.registries`.

**Comment by ialidzhikov** [comment]
/close
in favor of https://github.com/gardener/gardener/pull/10050

**Pull request closed without merging**